externalcreds/creds.go

```diff
@@ -10,6 +11,29 @@ func Set(opts *aws.Config) {
 	externalConfig = opts
 }
 
-func Get() *aws.Config {
-	return externalConfig
+func Get(region string) *session.Session {
+	if externalConfig == nil {
+		return session.Must(
+			session.NewSessionWithOptions(
+				session.Options{
+					SharedConfigState: session.SharedConfigEnable,
+					Config: aws.Config{
+						Region: aws.String(region),
+					},
+				},
+			),
+		)
+	} else {
+		return session.Must(
+			session.NewSessionWithOptions(
+				session.Options{
+					SharedConfigState: session.SharedConfigEnable,
+					Config: aws.Config{
+						Region:      aws.String(region),
+						Credentials: externalConfig.Credentials,
+					},
+				},
+			),
+		)
+	}
```

Comment:
I think this can be DRY'd up further.

E.g., you can do something like:

```go
config := aws.Config{Region: aws.String(region)}
if externalConfig != nil {
  config.Credentials = externalConfig.Credentials
}
return session.Must(
	session.NewSessionWithOptions(
		session.Options{
			SharedConfigState: session.SharedConfigEnable,
			Config: config,
		},
	),
)
```